cli/fmt.rs

```diff
@@ -22,16 +19,12 @@ use std::path::Path;
 use std::path::PathBuf;
 use std::time::Instant;
 
-lazy_static! {
-  static ref TYPESCRIPT_LIB: Regex = Regex::new(".d.ts$").unwrap();
-  static ref TYPESCRIPT: Regex = Regex::new(".tsx?$").unwrap();
-  static ref JAVASCRIPT: Regex = Regex::new(".jsx?$").unwrap();
-}
-
 fn is_supported(path: &Path) -> bool {
-  let path_str = path.to_string_lossy();
-  !TYPESCRIPT_LIB.is_match(&path_str)
-    && (TYPESCRIPT.is_match(&path_str) || JAVASCRIPT.is_match(&path_str))
+  if let Some(ext) = path.extension() {
+    ext == "ts" || ext == "tsx" || ext == "js" || ext == "jsx"
+  } else {
+    false
+  }
```

Comment:
dprint will break on `.d.ts` files and this change doesn't handle `.d.ts` properly:
```
let p = PathBuf::from("./a/b/c/asf.d.ts");
eprintln!("{:?}", p.extension());
// ts
```

Reply:
fixed